src/components/Config.tsx

```diff
@@ -159,44 +168,48 @@ const YamlTreeNode = ({ root, key, parentKey, value, separator }) => {
-        const hash = createUrlHash(parentKey, key);
+        const hash = createUrlHash(parentKey, name);
         if (window.location.hash === `#${hash}`) {
             scrollIntoView(hash);
         }
-    }, [key]);
+    }, [name]);
 
     return (
-        <div key={key} className={`highlight-config-node`} style={{ paddingLeft: `${root ? 0 : INDENT_SIZE}px` }} id={createUrlHash(parentKey, key)}>
+        <div key={name} className={`highlight-config-node`} style={{ paddingLeft: `${root ? 0 : INDENT_SIZE}px` }} id={createUrlHash(parentKey, name)}>
             <div className={`config-auxiliary-node`} style={{display: "inline-flex"}}>
-                {parseItalics(key)}{removeTrailingSpaces(separator)}
+                {parseItalics(name)}{removeTrailingSpaces(separator)}
             </div>
-            <a className={`config-anchor with-value-active-color hash-link`} href={`#${createUrlHash(parentKey, key)}`} onClick={handleClick}></a>
-            {renderYamlData(value, parentKey ? createUrlHash(parentKey, key) : parseUrlHash(key), false, separator)}
+            <a className={`config-anchor with-value-active-color hash-link`} href={`#${createUrlHash(parentKey, name)}`} onClick={handleClick}></a>
+            {renderYamlData(value, parentKey ? createUrlHash(parentKey, name) : parseUrlHash(name), false, separator, showAllDescriptions)}
         </div>
     );
 };
 
-const renderYamlData = (data, parentKey, root = false, separator) => {
-    const renderedNodes = [];
+const renderYamlData = (data, parentKey, root = false, separator, showAllDescriptions) => {
+    const renderedNodes: JSX.Element[] = [];
 
     for (const [key, value] of Object.entries(data)) {
-        if (typeof value === 'object') {
+        if (typeof value === 'object' && value !== null) {
             if ('default' in value || 'description' in value) {
-                renderedNodes.push(<YamlNodeWithDescription key={key} name={key} parentKey={parentKey} node={value} root={root} separator={separator} />);
+                renderedNodes.push(<YamlNodeWithDescription key={key} name={key} parentKey={parentKey} node={value} root={root} separator={separator} showAllDescriptions={showAllDescriptions} />);
             } else {
-                renderedNodes.push(YamlTreeNode({ root, key, parentKey, value, separator }));
+                renderedNodes.push(<YamlTreeNode root={root} key={key} name={key} parentKey={parentKey} value={value} separator={separator} showAllDescriptions={showAllDescriptions} />);
             }
         }
     }
 
     return renderedNodes;
 };
 
-export default function Config({ data, separator = ': ' }) {
+export default function Config({ data, separator = ': ', showDescriptions = false}) {
+    const [showAllDescriptions, setShowAllExpanded] = useState(showDescriptions);
     let ymlData = yaml.load(data);
     return (
         <div>
-            <pre>{renderYamlData(ymlData, '', true, separator)}</pre>
+            <pre className='codeblock'>
```

Comment:
codeblock as a classname is a bit generic here, considering its only position relative, maybe configblock?

Reply:
The sole purpose of the "position relative" class is to make it possible for a "position absolute" class to position correctly in whatever element it resides in. I originally renamed it to `config-button-parent` before changing my mind and naming it `relative-container` since it could be useful in other places in the future.

Reply:
Just realized I could also call it `config-container` :man_shrugging: